lib/driver.js

```diff
@@ -762,11 +762,9 @@ class XCUITestDriver extends BaseDriver {
       // local helper for the two places we need to uninstall wda and re-start it
       const quitAndUninstall = async (msg) => {
         this.log.debug(msg);
-        const noUninstallCap = ['webDriverAgentUrl', 'usePreinstalledWDA']
-          .find((x) => Boolean(this.opts[x]));
-        if (noUninstallCap) {
+        if (!this.isXcodebuildNeeded()) {
           this.log.debug(
-            `Not quitting/uninstalling WebDriverAgent since ${noUninstallCap} capability is provided`,
+            `Not quitting/uninstalling WebDriverAgent since ${!this.isXcodebuildNeeded()} capability is provided`,
```

Comment:
here we actually need to put capability names :)
maybe move them into a constant, which could be shared between this message and the isXcodebuildNeeded helper

Reply:
Wow, yea lol